### DISCUSSION TITLE
(pipelines): build failed during building `aws-cdk`, no export for `CredentialProviderSource`

### DISCUSSION THREAD

**Discussion by Mr-istov** [aws/aws-cdk#19495] [Q&A]
### What is the problem?

While working on some features on the `pipelines` module locally, the build failed when it tried to build `aws-cdk`.
This seems to be a regression from this PR https://github.com/aws/aws-cdk/pull/19324 from @rix0rrr.

After adding the `CredentialProviderSource` to the `export` members of the `credentials.ts` file, the build has succeeded.
I can provide a PR fix for this.

### Reproduction Steps

Try to build the `pipelines` package.

```
cd packages/@aws-cdk/pipelines
../../../scripts/buildup
```

### What did you expect to happen?

The build for the `pipelines` module to succeed.

### What actually happened?

Error from build:

```
/aws-cdk/packages/aws-cdk: yarn build  (57 remaining)
yarn run v1.22.17
$ cdk-build
lib/plugin.d.ts(1,10): error TS2305: Module '"./api/aws-auth/credentials"' has no exported member 'CredentialProviderSource'.
```

### CDK CLI Version

2.17.0

### Node.js Version

16.6.2

### OS

macOS Monterey

### Language

Typescript

**Comment by rix0rrr** [comment]
The fact that `master` still builds proves that there is not an inherent problem in the source or the build:

![image](https://user-images.githubusercontent.com/524162/159267968-f4503e84-9121-4993-bf81-00cf67ef6fe1.png)

You have old build artifacts lying around that are interfering with your build. Do a `git clean`, or as a shortcut run `scripts/clean-stale-files.sh`.
